### PR TITLE
Map: Add Map onBoundsChange prop and Popup onClose, onOpen, maxWidth and closeButton props

### PR DESCRIPTION
**Pull request by peterhass**
https://jira.nebenan.de/browse/CORE-6893

### CODE REVIEW

File: packages/map/src/popup/hooks.jsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { useCallback, useEffect, useRef } from 'react';
+
+// TODO: extract to hocs package?
```
Comment (peterhass) [inline packages/map/src/popup/hooks.jsx:3]: https://github.com/goodhood-eu/nebenan-react-hocs/pull/22

File: packages/map/src/popup/index.jsx [hunk 2]
```diff
@@ -45,6 +67,7 @@ const Popup = ({
 Popup.defaultProps = {
   offsetX: 0,
   offsetY: 0,
+  closeButton: true,
```
Comment (suprMax) [inline packages/map/src/popup/index.jsx:70]: `closable`?

**Review by suprMax — APPROVED** [review]

File: packages/map/src/popup/index.jsx (hunk 2 above)
Comment (peterhass) [reply packages/map/src/popup/index.jsx:70]: closeButton only shows the close button itself, the popup is always closable

File: packages/map/src/map/hooks.jsx [hunk 3]
```diff
@@ -64,14 +74,22 @@ export const useMapInstance = (nodeRef, options) => {
       setMap(map);
     };
 
+    const handleBoundsChange = (e) => {
+      const isCausedByUser = Boolean(e.originalEvent);
+      if (!isCausedByUser) return;
+
+      onBoundsChange(map.getBounds().toArray());
```
Comment (atomiomi) [inline packages/map/src/map/hooks.jsx:81]: Maybe pass bounds instead of array? There can be cases where you will need bounds

**Review by atomiomi — APPROVED** [review]

File: packages/map/src/map/hooks.jsx (hunk 3 above)
Comment (peterhass) [reply packages/map/src/map/hooks.jsx:81]: I thought about it but `bounds` is an object with many methods and I wouldn't want to have our apps rely on implementation details of this package. Let's say we change the underlying map library, then we would need to keep supporting the mapbox `bounds` object api.

**Review by suprMax — APPROVED** [review]
SGFMA

**Pull request merged**